hbp_service_client/storage_service/client.py

```diff
@@ -0,0 +1,239 @@
+        '''
+
+        self.__validate_storage_path(path)
+        entity = self.api_client.get_entity_by_query(path=path)
+        assert entity['entity_type'] == 'file'
+
+        signed_url = self.api_client.get_signed_url(entity['uuid'])
+        response = self.api_client.download_signed_url(signed_url)
+
+        with open(target_path, "wb") as output:
+            for chunk in response.iter_content(chunk_size=1024):
+                output.write(chunk)
+
+    def exists(self, path):
+        '''Check if a certain path exists in the storage service.
+
+        Args:
+            path (str): The path to be checked
+
+        Returns:
+            True if the path exists, False otherwise
+
+        Raises:
+            StorageArgumentException: Invalid arguments
+            StorageForbiddenException: Server response code 403
+            StorageNotFoundException: Server response code 404
+            StorageException: other 400-600 error codes
+        '''
+        self.__validate_storage_path(path)
+        try:
+            metadata = self.api_client.get_entity_by_query(path=path)
+        except StorageNotFoundException:
+            return False
+
+        return metadata and 'uuid' in metadata
+
+    def get_parent(self, path):
+        '''Get the parent entity of the entity pointed by the given path.
+
+        Args:
+            path (str): The path of the entity whose parent is needed
+
+        Returns:
+            A JSON object of the parent entity if found.
+
+        Raises:
+            StorageArgumentException: Invalid arguments
+            StorageForbiddenException: Server response code 403
+            StorageNotFoundException: Server response code 404
+            StorageException: other 400-600 error codes
+        '''
+        self.__validate_storage_path(path, projects_allowed=False)
+        path_steps = [step for step in path.split('/') if step]
+        del path_steps[-1]
+        parent_path = '/{0}'.format('/'.join(path_steps))
+        return self.api_client.get_entity_by_query(path=parent_path)
+
+    def mkdir(self, path):
+        '''Create a folder in the storage service pointed by the given path.
+
+        Args:
+            path (str): The path of the folder to be created
+
+        Returns:
+            None
+
+        Raises:
+            StorageArgumentException: Invalid arguments
+            StorageForbiddenException: Server response code 403
+            StorageNotFoundException: Server response code 404
+            StorageException: other 400-600 error codes
+        '''
+        self.__validate_storage_path(path, projects_allowed=False)
+        parent_metadata = self.get_parent(path)
+        self.api_client.create_folder(path.split('/')[-1], parent_metadata['uuid'])
+        #no return necessary, function succeeds or we would have thrown an exception before this point.
+
+    def upload_file(self, local_file, dest_path, mimetype, md5check=False):
```

Comment:
We can remove the md5check parameter as it's not used

Reply:
Well spotted. Removed.